### PR TITLE
DYN-4731-Ghost-Warning

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Fix the ghost warnings appearing when opening graphs.
I added a new constructor for the InfoBubbleViewModel class so we can pass the NodeViewModel and set the TopBottom and BottomRight positions of the warnings.
Also I changed to internal the methods NodeViewModel.GetTopLeft() and NodeViewModel.GetBotRight() so we have access to them in the InfoBubbleViewModel.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Fix the ghost warnings appearing when opening graphs.

### Reviewers

@QilongTang

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This is a GIF showing the functionality BEFORE my fix.
GhostWarningError

**Comment by RobertGlobant20** [comment]
This is a GIF showing the functionality AFTER my fix.
GhostWarningFixed

File: src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs [hunk 1]
```diff
@@ -465,6 +465,32 @@ public void OnRequestAction(InfoBubbleEventArgs e)
...
+        {
+            InitializeInfoBubble(dynamoViewModel);
+        }
+
+        /// <summary>
+        /// This constructor will update the TopBottom and BottomRight positions of the warnings since the beginning.
+        /// </summary>
+        /// <param name="dynamoViewModel"></param>
+        /// <param name="nodeViewModel">This parameter will be used to get the TopBottom and BottomRight positions of the NodeModel</param>
+        public InfoBubbleViewModel(DynamoViewModel dynamoViewModel, NodeViewModel nodeViewModel)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs:477]: `DynamoViewModel` was made public property on NodeViewModel so this function can be simplified to accept only one param

**Review by QilongTang — COMMENTED** [review]
One comment, then LGTM

File: src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs]: @QilongTang fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/13015/commits/ef41a1d4192da420f31d4dcdd78d26ef955e77ef

**Review by zeusongit — APPROVED** [review]

File: src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs [hunk 2]
```diff
@@ -465,6 +465,32 @@ public void OnRequestAction(InfoBubbleEventArgs e)
...
+        /// This constructor will update the TopBottom and BottomRight positions of the warnings since the beginning.
+        /// </summary>
+        /// <param name="dynamoViewModel"></param>
+        /// <param name="nodeViewModel">This parameter will be used to get the TopBottom and BottomRight positions of the NodeModel</param>
+        public InfoBubbleViewModel(NodeViewModel nodeViewModel)
+        {
+
+            InitializeInfoBubble(nodeViewModel.DynamoViewModel);
+
+            if (nodeViewModel != null)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs:482]: Should we move this check above the line above? I think param null check should be first of anything

**Review by QilongTang — COMMENTED** [review]
One more comment, then LGTM

File: src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs:479]: @QilongTang check moved one line above.
https://github.com/DynamoDS/Dynamo/pull/13015/commits/d18b0bc9afbe9f5abbdfefcbc8198dc32622b2ca

**Review by QilongTang — APPROVED** [review]

**Review by reddyashish — APPROVED** [review]

**Comment by QilongTang** [comment]
@RobertGlobant20 Going to merge this, please also cherry-pick to RC2.15.0_master branch

**Pull request merged**